client/tool/source/Main.cpp

```diff
@@ -133,22 +157,33 @@ static void stopCommand(const std::shared_ptr<IDobbyProxy>& dobbyProxy,
         }
     }
 
+    // Register an event listener to monitor for the container stop
+    promise = std::promise<bool>();
```

Comment:
Here there could be a problem if 2 `close` containers calls were done in very short time, as promise is global variable it could be override by second use. Furthermore `future.get` can be done only once (you would get "`std::future_error: No associated state`" if you try get more than once) - this is one more reason to use` future.wait()` for this.
We could solve this problem by creating additional mutex for promise and acquire it here (after making sure that it is valid close parameters) This way we would make sure that only one close is called at a time so global promise is safely connected to this future.